crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs

```diff
@@ -0,0 +1,80 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::JsStringLiteralExpression;
+use biome_rowan::AstNode;
+use regex::Regex;
+use std::sync::LazyLock;
+
+pub static OCTAL_ESCAPE_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r#"^(?:[^\\]|\\.)*?\\([0-7]{1,3})"#).unwrap()
+});
```

Comment:
Hi! Thanks for the quick PR!

Regexes have a large overhead. I was expecting an implementation similar to [no_useless_esapce_in_regex](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/lint/nursery/no_useless_escape_in_regex.rs#L58).

Reply:
ok, i will update this

Reply:
updated~